main/src/library/pkger/Manifest.flix

```diff
@@ -0,0 +1,287 @@
+
+        match dependencies {
+            case Some((version, dependencies)) => {
+                DepNode(package, version, dependencies |> mapWithKey((p, b) -> lockPackageHelper(p, b)) |> valuesOf)
+            }
+            case None => {
+                // TODO: No packages with that name and version combo
+                ???
+            }
+        }
+    }
+
+    def flattenDependencyTree(tree: DependencyTree): List[(String, SemVer.SemVer, List[(String, SemVer.SemVer)])] = {
+        match tree {
+            case DepNode(package, version, dependencies) => {
+                let x = List.flatMap(flattenDependencyTree, dependencies);
+                let flatDep = List.flatMap(match (p, v, d) -> (p, v) :: d, x);
+                (package, version, flatDep) :: x
+            }
+        }
+    }
+
+    // Resolve dependency tree and create a list of all dependencies and their transitive dependencies
+    pub def lockPackage(self: Package): List[LockedPackage] = {
+        DepNode( self.name
+               , self.version
+               , Map.toList(self.depends) |> List.map(match (p, b) -> lockPackageHelper(p, b))
+               )
+            |> flattenDependencyTree
+            |> List.map(match (package, version, dependencies) -> newLockedPackage(package, version, List.toMap(dependencies)))
+    }
+
+    // Create a new package. Everything but the name of the package can be populated with sensible defaults from the environment
+    pub def initPackage(name: String): /* Package */ Result[Path, ErrorFull] & Impure = {
+        let getProjectDir = (base) -> Path.add(base, name);
+
+        // Project directories
+        let getBuildDir = (base) -> Path.add(getProjectDir(base), "build");
+        let getSourceDir = (base) -> Path.add(getProjectDir(base), "src");
+        let getTestDir = (base) -> Path.add(getProjectDir(base), "test");
+
+        // Top-level files
+        let getPackageFile = (base) -> Path.add(getProjectDir(base), "package.sn");
+        let getHistoryFile = (base) -> Path.add(getProjectDir(base), "HISTORY.md");
+        let getLicenseFile = (base) -> Path.add(getProjectDir(base), "LICENSE.md");
+        let getReadmeFile = (base) -> Path.add(getProjectDir(base), "README.md");
+
+        // Source files
+        let getMainSourceFile = (base) -> Path.add(getSourceDir(base), "Main.flix");
+        let getMainTestFile = (base) -> Path.add(getTestDir(base), "TestMain.flix");
+
+        let newFile = (content) -> (path) -> {
+            Path.writeLines(path, Array.toList(content))
+        };
+
+        let workingDir = Path.workingDirectory();
+
+        let files = Array.toList([getPackageFile, getHistoryFile, getLicenseFile, getReadmeFile, getMainSourceFile, getMainTestFile]);
+        let dirs = Array.toList([getBuildDir, getSourceDir, getTestDir]);
+
+        // Ensure that none of the paths the init process creates are currently present on the system
+        let pathExists = [files, dirs]
```

Comment:
You probably want to compute the paths that already exists and then report them as errors.

Reply:
This is done in the latest version!